otelcollector/configuration-reader-builder/main.go

```diff
@@ -209,6 +209,7 @@ func taHealthHandler(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(status)
 		fmt.Fprintln(w, message)
 	}
+	defer resp.Body.Close()
```

Comment:
Thanks Rashmi, I've seen in the past when `http.Get` returns an error then `resp` is nil and `defer resp.Body.Close()` causes a segfault. Could we add `if resp != nil` around this?

Reply:
Thanks @gracewehner. Fixed.

Reply:
Thanks @gracewehner. Fixed